### DISCUSSION TITLE
The clock generation in the TB simulation using Python is slow

### DISCUSSION THREAD

**Discussion by jiao-x** [cocotb/cocotb#3810] [Q&A]
* the cocotb version：1.8.1
* the operating system and version ：centos7(64 bit),
* the simulator and version ：xcelium 22.03-s006,
* the Python version：python3.9  ,self install

The Python-based testbench written in cocotb is significantly slower compared to the testbench written in SystemVerilog, especially in cases where the clock frequency is higher.

**Comment by marlonjames** [comment]
https://github.com/cocotb/cocotb/issues/1676 has some recommendations on performance improvements.

It is highly recommended to generate your clock(s) in an HDL testbench wrapper module and use that as your top level.
